api/http/Server.go

```diff
@@ -91,13 +91,27 @@ func createMultiplexer(h1 core.CheckPermissionServer, h2 core.LicenseServiceServ
 		return nil, err
 	}
 
-	handler := cors.New(cors.Options{
+	chain := createChain(logMiddleware, corsMiddleware).then(mux)
+
+	return chain, nil
+}
+
+func corsMiddleware(h http.Handler) http.Handler {
+	return cors.New(cors.Options{
 		AllowedMethods:   []string{http.MethodHead, http.MethodGet, http.MethodPost, http.MethodPatch, http.MethodPut, http.MethodDelete, http.MethodOptions},
 		AllowedHeaders:   []string{"Accept", "ResponseType", "Content-Length", "Accept-Encoding", "Authorization", "Content-Type"},
 		AllowCredentials: true,
 		MaxAge:           300,
 		Debug:            true,
-	}).Handler(mux)
+	}).Handler(h)
+}
+
+func logMiddleware(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		glog.V(0).Infof("Request incoming: %s %s", r.Method, r.RequestURI)
+		glog.V(1).Infof("Request dump: %+v", *r)
 
-	return handler, nil
+		h.ServeHTTP(w, r)
```

Comment:
is it right that we call ServeHTTP in every middleware? Shouldn't that only be called at the end?

Reply:
If you are defining if with an inner HandlerFunc, then yes. If you click into the cors Handler, you'll see the same in there. This is what drives the chaining of the functions: each function calls the next. If you don't call it, you're basically terminating the request flow.

Reply:
Thanks for the clarification! Good to know :) I was a bit worried this would only work in HTTP mode now, but it is exactly what you said, the actual server is still started when we call either `http.ListenAndServe` or `http.ListenAndServeTLS` in Server.go lines 51 / 39. 

Just checked it locally using the generated tls certs:
![Screenshot 2023-04-20 at 12 46 59](https://user-images.githubusercontent.com/89905860/233343867-f83e322a-db90-4a3c-9ef5-a9cbf48e8c8f.png)
 
so all in all: LGTM, nice!

btw: don't worry too much about the CORS logger, that's for now only because we set it to debug mode.